cmd/root_test.go

```diff
@@ -296,17 +314,248 @@ func TestNewCommandArguments(t *testing.T) {
 
 	for _, tc := range tcs {
 		t.Run(tc.desc, func(t *testing.T) {
-			c := NewCommand()
-			// Keep the test output quiet
-			c.SilenceUsage = true
-			c.SilenceErrors = true
-			// Disable execute behavior
-			c.RunE = func(*cobra.Command, []string) error {
-				return nil
+			c, err := invokeProxyCommand(tc.args)
+			if err != nil {
+				t.Fatalf("want error = nil, got = %v", err)
+			}
+
+			if got := c.conf; !cmp.Equal(tc.want, got) {
+				t.Fatalf("want = %#v\ngot = %#v\ndiff = %v", tc.want, got, cmp.Diff(tc.want, got))
 			}
-			c.SetArgs(tc.args)
+		})
+	}
+}
+
+func TestNewCommandWithEnvironmentConfigPrivateFields(t *testing.T) {
+	tcs := []struct {
+		desc string
+		// sets an env var before the test, and returns a cleanup function
+		setEnv  func() func()
+		isValid func(cmd *Command) bool
+	}{
+		{
+			desc: "using the disable traces envvar",
+			setEnv: func() func() {
```

Comment:
**Optional** Every testcase has the same 3 lines of code in `setEnv`. It might be easier to read if you used a data field instead of a function. For example: 

```
{
	desc: "using the telemetry sample rate envvar",
	envName:  "CLOUD_SQL_TELEMETRY_SAMPLE_RATE",
	envValue: "500",
	isValid: func(cmd *Command) bool {
		return cmd.telemetryTracingSampleRate == 500
	},
}
```

Or like this to support multiple values:
```
{
	desc: "using the telemetry sample rate envvar",
	env:  map[string]string {
		"CLOUD_SQL_TELEMETRY_SAMPLE_RATE":"500",
	}
	isValid: func(cmd *Command) bool {
		return cmd.telemetryTracingSampleRate == 500
	},
}
```

Also applies to the `TestNewCommandWithEnvironmentConfigInstanceConnectionName` and `TestNewCommandWithEnvironmentConfig` test cases.

Reply:
Good idea. I'll update this.